tests/unit/amlb/datasets/file/test_file_dataloader.py

```diff
@@ -0,0 +1,345 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from amlb.resources import from_config
+from amlb.data import DatasetType
+from amlb.datasets.file import FileLoader
+from amlb.utils import Namespace as ns
+
+here = os.path.realpath(os.path.dirname(__file__))
+res = os.path.join(here, 'resources')
+
+
+@pytest.fixture(autouse=True)
+def file_config():
+    return from_config(
+        ns(
+            input_dir="my_input",
+            output_dir="my_output",
+            user_dir="my_user_dir",
+            root_dir="my_root_dir",
+        )
+    ).config
+
+
+@pytest.fixture()
+def file_loader(tmpdir):
+    return FileLoader(cache_dir=tmpdir)
+
+
+@pytest.mark.use_disk
+def test_load_binary_task_csv(file_loader):
+    ds_def = ns(
+        train=os.path.join(res, "kc2_train.csv"),
+        test=os.path.join(res, "kc2_test.csv"),
+        target="problems"
+    )
+    ds = file_loader.load(ds_def)
+    assert ds.type is DatasetType.binary
```

Comment:
These tests have a lot of duplicate code. I think it makes sense to create functions to re-use, e.g. `kc2_matches_expectations(ds)` that test the dataset on `kc2` characteristics (and similar for other datasets)?

Reply:
I know, there is some duplicated code but there are also some small differences between each test: I will try to improve this.

Reply:
by the way the file loading support still has several quirks: and the main benefit of those tests is to list many of them.